spring-kafka/src/main/java/org/springframework/kafka/core/KafkaTemplate.java

```diff
@@ -274,13 +274,17 @@ public boolean isTransactional() {
 		}
 
 		this.producers.set(producer);
+		boolean commitFailed = false;
 		try {
 			T result = callback.doInOperations(this);
+			commitFailed = true;
```

Comment:
Huh?

The variable name somehow doesn't reflect reality.
Isn't there a special exception to catch after `producer.commitTransaction()` failure?

Reply:
What do you mean; if we get an exception after this point, it's the commit that failed. I don't know all the possible exceptions from a commit.

Reply:
Yes, I see the logic and I agree about exceptions in the client.
Only what I don't feel comfort is the name of the variable. 
Since it is going to be `true` even if commit is not failed.

Reply:
Seems ok to me 😄 

Would you prefer `callbackWasSuccessful = true` and `if (!callbackWasSuccessful)` ?